bokeh/plotting/contour.py

```diff
@@ -17,23 +17,73 @@
 # Imports
 #-----------------------------------------------------------------------------
 
+# Standard library imports
+from typing import Any, Dict, Optional, Sequence, Tuple, Union
+
 # External imports
 import numpy as np
 
+# Bokeh imports
+from ..models.renderers import ContourRenderer
+
 #-----------------------------------------------------------------------------
 # Globals and constants
 #-----------------------------------------------------------------------------
 
 __all__ = (
     'contour_data_dicts',
     'contour_coords',
+    'from_contour',
 )
 
 #-----------------------------------------------------------------------------
 # General API
 #-----------------------------------------------------------------------------
 
-def contour_coords(x, y, z, levels, want_fill, want_line):
+def from_contour(
+    x: Optional[np.ndarray],
+    y: Optional[np.ndarray],
+    z: Union[np.ndarray, np.ma.MaskedArray],
```

Comment:
Perhaps `npt.ArrayLike` would be a better option (I'm not sure).

Reply:
`npt.ArrayLike` is what I would like to use, but it is only available from `numpy` 1.20 onwards.

Reply:
We already use this across the codebase. Just put the import under `TYPE_CHECKING` (from `typing` module), and it won't affect the required run-time version of numpy (which currently is `>=1.11`).